File: java/src/test/java/problems1/ValidParenthesesTest.java
package problems1;

import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ValidParenthesesTest extends SampledTest {
    @Override
    public String getFileName() {
        return "valid_parentheses.json";
    }

    @Test
    public void test() {
        for (ValidParentheses solution : ValidParentheses.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isValid(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, Boolean> {
        public static class Input {
            public String s;
        }
    }
}

File: typescript/__tests__/problems1_100/valid-parentheses.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/valid-parentheses';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    s: string,
  },
  output: boolean,
  message: string,
}

describe(
    'ValidParentheses',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution => {
                  getSamples().forEach((sample: Sample) => {
                    strictEqual(
                        solution.isValid(sample.input.s),
                        sample.output,
                        sample.message,
                    );
                  });
                }),
        ),
);
